scalyr_agent/builtin_monitors/linux_process_metrics.py

```diff
@@ -113,6 +113,12 @@ class Metric(object):
     """
     __slots__ = 'name', 'type', '_frozen'
 
+    def __hash__(self):
+        return hash(self.name) + hash(self.type)
```

Comment:
The purest in me says that you should multiple the second one by a prime number to get around periodicity in the sequence.  So,
`return hash(self.name) + 13 * hash(self.type)`